### DISCUSSION TITLE
Environment level filters

### DISCUSSION THREAD

**Discussion by yoav-sentinel** [pallets/jinja#1547] [Show and tell]
Hi

Is it possible to create jinja environment with default filters that will be added to all of the parameters automatically?

My specific use case is a requirement to add the following filter to all of my parameters (hundreds):
`'| replace('|','\|')'`
The pipe character needs to be escaped as I'm sending CEF messages where pipe is the delimiter unless escaped.

Another use case I had, is to add a default value to all parameters:
`'| default("", true)' `

It could be very useful to have environment level filters. For example:
```
env = jinja2.Environment(filters=[r'''default("this is default", true)''', r'''replace('|','\|')'''])
template = env.from_string(" {{ first_name }} {{ last_name }} ")
template.render({"first_name": "first"})

>> first this is default
```

**Comment by ThiefMaster** [comment]
I used a small unicode extension to apply a filter to everything some years ago, maybe it's a useful starting point for you: https://github.com/indico/indico/blob/v1.9.11.dev18/indico/web/flask/templating.py#L258

**Reply by yoav-sentinel** [reply]
Thanks @ThiefMaster. 
Currently string replace workaround do the trick for me as I only need to add a filter at the end of a parameter. It's pretty ugly though:

`replace('}}', r'''| replace("|", "\|") }}''')`
